frontend/components/ActivityDetails/InstallDetails/SoftwareUninstallDetailsModal/SoftwareUninstallDetailsModal.tsx

```diff
@@ -167,41 +167,60 @@ const SoftwareUninstallDetailsModal = ({
       ...DEFAULT_USE_QUERY_OPTIONS,
       // are 4xx errors other than 404 expected intermittently?
       retry: (failureCount, err) => err?.status !== 404 && failureCount < 3,
+      // Prevent any error UI with pending uninstall
+      enabled: uninstallStatus !== "pending_uninstall",
     }
   );
 
-  if (isLoading) {
-    return <Spinner />;
-  } else if (isError && error?.status === 404) {
+  const renderContent = () => {
+    if (isLoading) {
+      return <Spinner />;
+    } else if (isError && error?.status === 404) {
+      return (
+        <DataError
+          description="These uninstall details are no longer available."
+          excludeIssueLink
+        />
+      );
+    } else if (isError) {
+      return <DataError description="Close this modal and try again." />;
+    } else if (!uninstallResult && uninstallStatus !== "pending_uninstall") {
+      // FIXME: Find a better solution for this.
+      return <DataError description="No data returned." />;
+    }
+
     return (
-      <DataError
-        description="These uninstall details are no longer available."
-        excludeIssueLink
-      />
+      <div className={`${baseClass}__modal-content`}>
+        <StatusMessage
+          host_display_name={hostDisplayName || ""}
+          status={
+            (uninstallStatus || "pending_uninstall") as SoftwareUninstallStatus
+          }
+          softwareName={softwareName}
+          softwarePackageName={softwarePackageName}
+          timestamp={uninstallResult?.created_at}
+          isDUP={!!deviceAuthToken}
+          contactUrl={contactUrl}
+        />
+        {uninstallStatus !== "pending_uninstall" && (
```

Comment:
For next iteration: one check for `!== "pending_uninstall"` and maintaining [encapsulation of this "details section"](https://github.com/fleetdm/fleet/pull/31420/commits/f37b07e0d4816a6a68b522921d606b5a2f804f61#diff-9e4043ea5f0a3c4a14fac9bf75361e5288cf783ac914326967fbe8d32f8fbbe4L189) would  be  bit clearer. Okay for now for speed.

Reply:
100